l293x/l293x.go

```diff
@@ -61,44 +61,67 @@ func (d *Device) Stop() {
 // en is the PWM pin that controls the motor speed.
 type PWMDevice struct {
 	a1, a2 machine.Pin
-	en     machine.PWM
+	en     machine.Pin
+	pwm    *machine.TCC
```

Comment:
To make this driver portable, the l293x package should define a new `PWM` interface and use that. For an example, see:

https://github.com/tinygo-org/drivers/blob/8d776b9ebbf51f7b0b761e883394a5cf3553fc74/tone/tone.go#L7-L14

Reply:
Done.